In [1]:
import numpy as np
from LoadUnipen import LoadUnipen
from Testing.CLayer.loadData import load_dataset
import pandas as pd

## Partie Unipen

Pour Unipen on est censé retrouver des séquences de tailles 50. "The patterns were preprocessed by re-sampling them to 50 sequence elements"

In [2]:
X, y = LoadUnipen()

(15578,)


C:\Users\traz\Dropbox\M2\Stage\Python\DTW_Stage\LoadUnipen.py:99: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  X = np.asarray(X)


In [3]:
len(X)

15578

In [4]:
uni_len = {}
for ts in X:
    uni_len[len(ts)] = uni_len.get(len(ts), 0) + 1
dstbt = pd.DataFrame.from_dict(uni_len, orient='index', columns=['nb'])

In [5]:
dstbt["group"] = pd.qcut(dstbt.index, 10)
print(dstbt.groupby("group").sum())

                  nb
group               
(-0.001, 29.0]  6299
(29.0, 57.0]    4515
(57.0, 85.0]    3277
(85.0, 113.0]   1024
(113.0, 141.0]   211
(141.0, 169.0]    82
(169.0, 203.0]    58
(203.0, 246.0]    44
(246.0, 319.0]    38
(319.0, 868.0]    30


In [6]:
print(dstbt[dstbt.index==0])

     nb           group
0  4028  (-0.001, 29.0]


On voit que la taille des séries varie. Difficile de retrouver 13000 séries avec déjà 4028 qui sont vident (peut être un problème dans ma récupération des données!). 

In [7]:
nb_null = 0
for ts in X:
    if ts.mean()==0:
        nb_null +=1
print(nb_null)

C:\Users\traz\AppData\Local\Temp\ipykernel_20144\1464646224.py:3: RuntimeWarning: Mean of empty slice.
  if ts.mean()==0:
C:\Users\traz\Anaconda3\envs\stageenv\lib\site-packages\numpy\core\_methods.py:189: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)


208


## Partie ADL

Ici on doit trouver des sequences de 200. The time series were also sampled to 200 frames. "The time series were also sampled to 200 frames. The original experiment [4] uses a median filter, however we found it unnecessary and used the raw data for the experiment."

In [3]:
X_test, y_test, X_train, y_train = load_dataset("UCI", "ADL")

C:\Users\traz\Dropbox\M2\Stage\Python\DTW_Stage\Testing\CLayer\loadData.py:273: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  X_ = np.array(X_seq)
C:\Users\traz\Dropbox\M2\Stage\Python\DTW_Stage\Testing\CLayer\loadData.py:273: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  X_ = np.array(X_seq)


In [6]:
len(X_train)

600

In [8]:
adl_len = {}
for ts in X_train:
    adl_len[len(ts)] = adl_len.get(len(ts), 0) + 1
dstbt = pd.DataFrame.from_dict(adl_len, orient='index', columns=['nb'])

In [9]:
dstbt["group"] = pd.qcut(dstbt.index, 10)
print(dstbt.groupby("group").sum())

                  nb
group               
(124.999, 194.2]  84
(194.2, 272.2]    77
(272.2, 341.8]    51
(341.8, 380.4]    74
(380.4, 428.0]    77
(428.0, 475.6]    69
(475.6, 527.2]    53
(527.2, 658.4]    38
(658.4, 913.8]    38
(913.8, 3153.0]   39


In [12]:
print("Taille min: ", dstbt.index.min())
print("Taille max: ", dstbt.index.max())

Taille min:  125
Taille max:  3153


In [13]:
nb_null = 0
for ts in X_train:
    if ts.mean()==0:
        nb_null +=1
print(nb_null)

0
